spring-cloud-gcp-starters/spring-cloud-gcp-starter-core/src/main/java/org/springframework/cloud/gcp/core/GCPProperties.java

```diff
@@ -27,7 +28,7 @@
 
 	private String projectId;
 
-	private String jsonKeyFile;
+	private Resource credentialKeyFile;
```

Comment:
@viniciusccarvalho This definitely helps. I think the part I was missing is that in non-boot Spring, there is no `@ConfigurationProperties`. Consequently, using `GcpProperties` in `MessageHandler` would make it specific to boot. The part I'm still not completely sure on is `GcpProjectIdProvider`. Where would that be useful?